test/httpProblemDetailsFactory.test.ts

```diff
@@ -0,0 +1,24 @@
+import { HttpProblemDetailsFactory } from '../src';
+
+describe('given HttpProblemDetailsFactory', () => {
+  describe('when trying to get not exist type', () => {
+    it('throw an Error', () => {
+      const factory = new HttpProblemDetailsFactory();
+      expect(() => factory.get('nonExistedType')).toThrowError();
+    });
+  });
+
+  describe('when trying to get existed type', () => {
+    it('return instance of the problem based on the definition', () => {
+      const factory = new HttpProblemDetailsFactory();
+      const type = 'https://example.com/probs/out-of-credit';
+      factory.addDefinition({
+        status: 403,
+        type,
+        title: 'You do not have enough credit.',
+      });
+      const problem = factory.get(type);
```

Comment:
Thing to consider - should we `get` details from factory based on `type`? I might be wrong, but I see `type` as merely one of the attributes of detail. What's more, when we consider one of the previous remarks, there might be many different details with the same type - "about:blank". 

I think we need to find a different kind of id to `get` by